mlflow/models/evaluation/default_evaluator.py

```diff
@@ -139,8 +139,10 @@ def _extract_raw_model(model):
         return model_loader_module, None
 
 
-def _extract_predict_fn(model, raw_model):
+def _extract_predict_fn(model, raw_model, model_predict_fn=None):
     predict_fn = model.predict if model is not None else None
+    if model_predict_fn is not None:
+        predict_fn = model_predict_fn
```

Comment:
Can we add a docstring that indicates the precedence order and restructure this method so that the logic is if / elif / else? It's currently difficult to reason about what happens when `model_predict_fn` is specified and `raw_model` is also specified, etc.

Can we also add type hints to the function to indicate which parameters are optional?